docs/features/runs/sweep-syntax.md

```diff
@@ -6,11 +8,12 @@ Grid allows running [hyperparameter sweeps](https://www.grid.ai/what-are-hyperpa
 python main.py --layers 32 --learning_rate 0.01
 ```
 
-:::note
+<note>
 This page provides a list of arguments specific to running hyperparameter sweeps. For the full Grid CLI API reference, visit [this page. ](../../cli/api.md)
-:::
+</note>
 
 ## Python flags
+Under the hood we are using Argparse for parsing of Python flags and consequently inherit the same limitations it has. For example, there is a known limitation with parsing boolean values. See the official Argparse [documentation](https://docs.python.org/3/library/argparse.html#type). However, Grid still provides a way to easily pass in Boolean flags.
```

Comment:
We don't use argparse under the hood. 

The question raised in the ticket was "how can I use boolean flags with Grid Syntax for grid search"?

For instance, most scripts use the presence / absence of a flag to set a boolean value. e.g. `python script.py` versus `python script.py --use-advanced-feature`.  If you want to run grid search over the boolean value (some experiments with `--use-advanced-feature` and some without), how can you use [Grid Syntax](https://docs.grid.ai/features/runs/sweep-syntax) to achieve this?

Our current documentation is incomplete. If we try to run `grid run script.py --flag`, we don't run two experiments, we just run 1:
```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━┓
┃ Experiment                   ┃ flag ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━┩
│ gifted-mirzakhani-5668-exp0  │ True │
└──────────────────────────────┴──────┘
```

My suggestion was for the user to change how their script accepts boolean values. Instead of using the presence / abscence of a flag, parse `True` or `False` from the command line. That way you can use the existing Grid Syntax to perform grid search.

`grid run script.py --opt1 "[True, False]" --opt2 "[True, False]"`:
```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━┳━━━━━━━┓
┃ Experiment                ┃       Command ┃  opt1 ┃  opt2 ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━╇━━━━━━━┩
│ unyielding-cray-8074-exp0 │ bool-param.py │  True │  True │
│ unyielding-cray-8074-exp1 │ bool-param.py │  True │ False │
│ unyielding-cray-8074-exp2 │ bool-param.py │ False │  True │
│ unyielding-cray-8074-exp3 │ bool-param.py │ False │ False │
└───────────────────────────┴───────────────┴───────┴───────┘
```

I proposed an `argparse` solution in the ticket since I saw a python script at the end of the ticket details. In reality, it's up to the user to decide how to parse `True` and `False` instead of presence / abscence of a flag.

Reply:
I thought we did per - https://github.com/gridai/grid/blob/master/grid-backend/hyperparameter_parser/hyperparameter_parser/command.py

Reply:
@luca3rd though even if it is true it is probably unnecessary information. So I will remove it and just leave the comment of passing in 0/1 values as an option.

Reply:
Ah, yes, so we do use argparse. But I think the documentation could be clearer. I attached an example in my other comment.